### PR TITLE
fix(portal): detect changes for portal hostview while before attaching.

### PR DESCRIPTION
**Pull request by josephperrott**
#4318

### CODE REVIEW

**Comment by josephperrott** [comment]
@jelbourn ready for you to take a look

File: src/lib/core/portal/dom-portal-host.ts [hunk 1]
```diff
@@ -29,21 +29,17 @@ export class DomPortalHost extends BasePortalHost {
    */
   attachComponentPortal<T>(portal: ComponentPortal<T>): ComponentRef<T> {
     let componentFactory = this._componentFactoryResolver.resolveComponentFactory(portal.component);
-    let componentRef: ComponentRef<T>;
+    let componentRef = componentFactory.create(portal.injector || this._defaultInjector);
+    componentRef.hostView.detectChanges();
```
Comment (jelbourn) [inline src/lib/core/portal/dom-portal-host.ts]: Are you able to create a unit test that enforces this behavior?

**Comment by josephperrott** [comment]
Should be ready for another look.

I updated `MdDatepickerContent` to use the safe navigation operator.  This is necessary as the newly created `ComponentPortal<MdDatepicker>` instance is immediately change detected.  We need to guard against `undefined` and `null` since the inputs aren't guaranteed to be populated.

cc:@mmalerba as this is modifying the newly created `MdDatepicker`

**Comment by mmalerba** [comment]
LGTM for datepicker

**Comment by willshowell** [comment]
@jelbourn @mmalerba @josephperrott the CD changes made here are causing several regressions with the datepicker in popup view

1. Unable to start in year view
2. Start at date has no effect
3. "Prev month" button initially enabled when min date in current month
4. When setting min/max dates within current month, all days are enabled until month view is refreshed
5. Active date is not highlighted until month view is refreshed

ref #5065

**Pull request merged**